### DISCUSSION TITLE
Support for Seeed Studio XIAO nrf52840

### DISCUSSION THREAD

**Discussion by pdcook** [h2zero/n-able-Arduino#13] [Ideas]
I tried going about adding this myself, but after several hours of tracking down linking issues, I was only able to get sketches which _didn't_ use NimBLE to compile (they use a different version of freertos), which defeats the whole purpose.

In my understanding, it might be possible to just compile for the Generic nrf52840 (with some minor variant stuff for pinout) and just use the uploading scripts from [their core here](https://github.com/zhanggw7/Seeed_52840_Core/blob/main/README.md).

If this is something someone else feels comfortable doing, I would greatly appreciate it, but I'm also happy to do it myself if someone with experience in this sort of thing can provide some guidance and help me.

Cheers

**Comment by pdcook** [comment]
So compiling for `Generic nRF52840` (with the `variant` code from the Xiao put in place of the generic one) then uploading manually does not work. Rather, no errors occur in the compilation/uploading process, but the board does not boot.

**Comment by h2zero** [comment]
Sorry, I havent had time to look into this. I suspect there may be an issue with the pin mapping but I don't have a way to test that.

**Comment by babolcs** [comment]
I tried to do a compilation (A code which ran on xiao esp32c3, but got this result/error:
(I used the generic nrf52840 board (i use the xiao one)
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes
c:/users/szabo/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes

exec: "echo": executable file not found in %PATH%

Compilation error: exec: "echo": executable file not found in %PATH%

**Reply by h2zero** [reply]
That's strange, why is echo not available on your computer?

**Reply by alexthecoolman** [reply]
Hi, I try to do an integration of BTHome with XIAO Seeed nrf52840 with Your lib, same issue "exec: "echo": executable file not found in %PATH%" (via arduino ide)

**Reply by intakenick** [reply]
@h2zero I'm having the same issue when compiling for a generic NRF51822:
`exec: "echo": executable file not found in %PATH%`
